app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt

```diff
@@ -2831,6 +2871,8 @@ class ChatController(args: Bundle) :
 
         if (!CapabilitiesUtil.hasSpreedFeatureCapability(conversationUser, "delete-messages")) return false
 
+        if (!hasChatPermission) return false
```

Comment:
to be combined with the above check or adding a return when for all 3 cases

Reply:
i don't get the point yet.

you want a when condition like

when {
    condition1 -> ...
    condition2 -> ...
    condition3 -> ...
}

for the last 3 cases?

Reply:
used return when for more cases here..

Reply:
Yeah, not limited for 3 cases of course, but that is just what is visible in GH's review view, when not expanding the diff view ;)